addon/index.js

```diff
@@ -1,12 +1,13 @@
 import { getProviderFor } from './-private/get-provider-for';
 
-export function inject(key) {
+export function inject(key, opts) {
   return function decorator() {
     return {
       get() {
-        const provider = getProviderFor(this, key);
+        const fallback = opts ? opts.fallback : null;
```

Comment:
FYI made a small change here since deconstructing the fallback in the arguments was erroring if no object was passed (used to be `export function inject(key, { fallback }) { ... }`)

Reply:
Another slightly cleaner option is to provide a default value for the second parameter to avoid that destructuring-from-`undefined` problem

```javascript
export function inject(key, { fallback } = {}) {
```

Which is more or less the same as what you're doing here, just slightly more clean if you wanted to make that change.